read-sentinel-data-minimal.py

```diff
@@ -1,91 +1,152 @@
-            )
+# stack all timepoints together for each band
+images = {}
+for band, shape in zip(BANDS, IM_SHAPES):
+    stack = []
+    for fn in all_zips:
+        basepath = os.path.splitext(os.path.basename(fn))[0]
+        path = basepath + '/' + basepath + '_' + band + '.tif'
+        image = da.from_delayed(
+            ziptiff2array(fn, path), shape=shape, dtype=np.int16
         )
+        stack.append(image)
+    images[band] = da.stack(stack)
 
-with napari.gui_qt():
-    start = time.time()
-    v = napari.view_image(all_images[0], name=IM_SUFFIXES[0] + '_' + str(SCALES[0]), is_pyramid=False, scale=SCALES[0],
-        translate=(5, 5) if SCALES[0] == [20, 20, 0] else None)
-    end_1 = time.time()
 
-    # add remaining labels, making RGB visible and blending
-    for i, img in enumerate(all_images[1:], start=1):
-        if IM_SUFFIXES[i] == 'SRE_B2':
-            v.add_labels(img, name=IM_SUFFIXES[i] + '_' + str(SCALES[i]), is_pyramid=False, visible=True, scale=SCALES[1],
-            translate=(5, 5) if SCALES[i] == [20, 20, 0] else None,
-            colormap='blue',
-            blending='additive')
-        elif IM_SUFFIXES[i] == 'SRE_B3':
-            v.add_labels(img, name=IM_SUFFIXES[i] + '_' + str(SCALES[i]), is_pyramid=False, visible=True, scale=SCALES[1],
-            translate=(5, 5) if SCALES[i] == [20, 20, 0] else None,
-            colormap='green',
-            blending='additive')
-        elif IM_SUFFIXES[i] == 'SRE_B4':
-            v.add_labels(img, name=IM_SUFFIXES[i] + '_' + str(SCALES[i]), is_pyramid=False, visible=True, scale=SCALES[1],
-            translate=(5, 5) if SCALES[i] == [20, 20, 0] else None,
-            colormap='red',
-            blending='additive')
-        else:
-            v.add_labels(img, name=IM_SUFFIXES[i] + '_' + str(SCALES[i]), is_pyramid=False, visible=False, scale=SCALES[1],
-            translate=(5, 5) if SCALES[i] == [20, 20, 0] else None)
-    end_2 = time.time()
+colormaps = defaultdict(lambda: 'gray')
+for band in BANDS:
+    if band.endswith('B2'):
+        colormaps[band] = 'red'
+    elif band.endswith('B3'):
+        colormaps[band] = 'green'
+    elif band.endswith('B4'):
+        colormaps[band] = 'blue'
 
-    print("First image: ", end_1 - start)
-    print("Remaining layers: ", end_2 - end_1)
 
+with napari.gui_qt():
+    v = napari.Viewer()
+    times = []
+    visibles = []
+    for band, image in images.items():
+        colormap = colormaps[band]
+        blending = 'additive' if colormaps[band] != 'gray' else 'translucent'
+        visible = (band in {'SRE_B2', 'SRE_B3', 'SRE_B4'})
+        start = time.time()
+        v.add_image(
+            image,
+            name=band,
+            is_pyramid=False,
+            scale=SCALES[band],
+            translate=OFFSETS[band],
+            colormap=colormap,
+            blending=blending,
+            visible=visible,
+            contrast_limits=[-1000, 19_000],
```

Comment:
This too — sorry it didn't occur to me sooner!

Reply:
(might be worth tweaking these values to something more visible, as this gives pretty dark results on the standard color bands)